small/region.c

```diff
@@ -81,17 +81,26 @@ region_truncate(struct region *region, size_t used)
 						       slab.next_in_list);
 		if (slab->used > cut_size) {
 			/* This is the last slab to trim. */
+#ifndef NDEBUG
+			memset(rslab_data_end(slab) - cut_size, '#', cut_size);
+#endif
 			slab->used -= cut_size;
 			cut_size = 0;
+
 			break;
 		}
+#ifndef NDEBUG
+		memset(rslab_data(slab), '#', slab->used);
+#endif
 		cut_size -= slab->used;
 		/* Remove the entire slab. */
 		slab_list_del(&region->slabs, &slab->slab, next_in_list);
 		slab_put(region->cache, &slab->slab);
 	}
 	assert(cut_size == 0);
 	region->slabs.stats.used = used;
+	if (small_unlikely(region->on_truncate != NULL))
+		region->on_truncate(region, used, region->opaque);
```

Comment:
Shouldn't we also invoke the callback on `region_free`?

Reply:
Done.